### PR TITLE
Add HTTP_PROXY/HTTP_PROXY_AUTH/HTTP_TIMEOUT to Popen() environment if set

### LINKED ISSUE #1159: iocage [upgrade|fetch] does not take proxy settings into account
Make sure to follow and check these boxes before submitting an issue! Thank you.

- [X] Supply `iocage --version`
```
# iocage --version
Version	FreeNAS 11.2-U6
```

- [X] Supply the commands used, along with any steps to recreate it.
- [X] Provide the output from the command you issued.
Run
```
# setenv | grep -i proxy
HTTPS_PROXY=http://<host>:<port>
HTTP_PROXY=http://<host>:<port>
### the following ones may be not relevant
http_proxy=http://<host>:<port>
https_proxy=http://<host>:<port>

# iocage upgrade <jail_with_older_release> -r 11.2-RELEASE
Looking up update.FreeBSD.org mirrors... 3 mirrors found.
Fetching metadata signature for 11.1-RELEASE from update1.freebsd.org... done.
Fetching metadata index... done.
Fetching 1 metadata patches. done.
Applying metadata patches... done.
Fetching 1 metadata files... done.
Inspecting system... done.

The following components of FreeBSD seem to be installed:
src/src world/base world/doc world/lib32

The following components of FreeBSD do not seem to be installed:
world/base-dbg world/lib32-dbg


Fetching metadata signature for 11.2-RELEASE from update1.freebsd.org... done.
Fetching metadata index... done.
Fetching 1 metadata patches. done.
Applying metadata patches... done.
Fetching 1 metadata files... done.
Inspecting system... done.
Fetching files from 11.1-RELEASE for merging... done.
Preparing to download files... done.
Fetching 42999 patches.....10....20....30....40....50....60 done.
Applying patches... done.
Fetching 46594 files... ....10... failed.
Traceback (most recent call last):
  File "/usr/local/bin/iocage", line 10, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/iocage_cli/upgrade.py", line 39, in cli
    ioc.IOCage(jail=jail, skip_jails=True).upgrade(release)
  File "/usr/local/lib/python3.6/site-packages/iocage_lib/iocage.py", line 1805, in upgrade
    root_path).upgrade_jail()
  File "/usr/local/lib/python3.6/site-packages/iocage_lib/ioc_upgrade.py", line 116, in upgrade_jail
    stdin=su.PIPE, env=self.upgrade_env)
RuntimeError: Error occured, jail not upgraded!
```
When inspecting the proxy-logs, no request could be seen from that activitiy.

Also fetching may fail:
```
# iocage fetch -r 11.2-RELEASE
Fetching: 11.2-RELEASE

Extracting: base.txz... 
Extracting: lib32.txz... 
Extracting: doc.txz... 
Extracting: src.txz... 

* Updating 11.2-RELEASE to the latest patch level... 
Looking up update.FreeBSD.org mirrors... 3 mirrors found.
Fetching metadata signature for 11.2-RELEASE from update2.freebsd.org... failed.
Fetching metadata signature for 11.2-RELEASE from update1.freebsd.org... failed.
Fetching metadata signature for 11.2-RELEASE from update4.freebsd.org... done.
Fetching metadata index... done.
Inspecting system... done.
Preparing to download files... done.
Fetching 773 patches.....10....20. done.
Applying patches... done.
Fetching 852 files...  failed.
No updates are available to install.
Run '/tmp/tmp3ecrkdus fetch' first.
```
Again no entries in the proxy-logs.

- [X] Supply what you expected the result or output to be
`iocage [fetch|upgrade]` should take `HTTP_PROXY*` settings into account and use them in the environment of `Popen` when calling sub-commands.

- [X] Checked that the problem has not already been fixed on `master` if using
 a stable release.
The current  source code still does not provide those settings for the sub-commands invoked

### PR DESCRIPTION
**Pull request by ableischwitz**
With that patch `iocage` is taking the proxy environment variables into account when invoking `freebsd-update.sh` as a sub-process.

Fixes #1159 

Make sure to follow and check these boxes before submitting a PR! Thank you.

- [X] Explain the feature
- [X] Read [CONTRIBUTING.md](https://github.com/iocage/iocage/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: iocage_lib/ioc_fetch.py [hunk 1]
```diff
@@ -866,6 +866,10 @@ def fetch_update(self, cli=False, uuid=None):
         path = '/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:'\
                '/usr/local/bin:/root/bin'
         fetch_env = {
+            **{
```
Comment (sonicaj) [inline iocage_lib/ioc_fetch.py]: We should ensure `http_proxy` as well and we also have `https_proxy`, so we should check that too

**Comment by skarekrow** [comment]
Sorry about that! Some git-fu backfired, can you please reopen this? Thanks!

**Comment by ableischwitz** [comment]
Didn't know how to re-open - also there seem to be a lot of changes made to master, so I rebased my patch to current master.
See the following pull-request: https://github.com/iocage/iocage/pull/1212

File: iocage_lib/ioc_fetch.py (hunk 1 above)
Comment (ableischwitz) [reply iocage_lib/ioc_fetch.py]: As far as I read the sources of freebsd-update, it does only consider those variables. IMHO no need to add others if they are not used. Also I had to re-create a pull-request: https://github.com/iocage/iocage/pull/1212

Comment (ableischwitz) [reply iocage_lib/ioc_fetch.py]: If you would like to follow the path: freebsd-update uses fetch to pull the images, fetch uses fetchXGet() for downloading and according to https://www.freebsd.org/cgi/man.cgi?apropos=0&sektion=3&query=fetchXGet&manpath=FreeBSD+7.0-current&format=html only HTTP_PROXY is used, while http_proxy is for compatibility reasons. Not sure if FTP_PROXY should be added though. AFAIK iocage only uses HTTP nowadays.

**Pull request closed without merging**